File: BOJ/단계별 문제/7. 문자열/10. 그룹 단어 체커/set.py
T = int(input())
ans = T

for _ in range(T):
    word = input()
    appeared = set()
    p = ''

    for i, x in enumerate(word):
        if x not in appeared:
            appeared.add(x)

        elif x != p:
            ans -= 1
            break

        p = x

print(ans)

File: BOJ/단계별 문제/7. 문자열/10. 그룹 단어 체커/set.cpp
#include <iostream>
#include <set>
using namespace std;

int main() {
    int T, ans;

    cin >> T;
    ans = T;

    while (T--) {
        string word;
        set<char> appeared;

        cin >> word;

        for (int i=0; i<word.length(); i++) {
            if (appeared.find(word[i]) == appeared.end())
                appeared.insert(word[i]);

            else if (word[i-1] != word[i]) {
                ans--;
                break;
            }
        }
    }

    cout << ans;
    return 0;
}